tests/integration/azure/test_resources.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+from unittest.mock import create_autospec
```

Comment:
Integration tests have to use real objects, not the ones created by magicmock. This is a unit test